src/basics.md

```diff
@@ -470,6 +471,48 @@ fn main() {
 }
 ```
 
+[ex-phone]: #ex-phone
+<a name="ex-phone"></a>
+## Extract phone numbers from text
+
+Attempts to process a user input as a US phone number.  Essentially, the
+expression is checking that there are 7 digits that come in groups of 3 3 4,
+removing any extra characters.  The phone number is then formatted for display.
+
+```rust
+# #[macro_use]
+# extern crate error_chain;
+extern crate regex;
+
+use std::io::stdin;
+use regex::Regex;
+#
+# error_chain!{
+#     foreign_links {
+#         Regex(regex::Error);
+#         Io(std::io::Error);
+#     }
+# }
+
+fn run() -> Result<()> {
+    println!("Input a phone number");
+    let mut phone_input = String::new();
+    // Regular expression is looking for 3 groups of digits, capturing them
+    let re = Regex::new(r".*([2-9]\d{2}).*(\d{3}).*(\d{4})")?;
```

Comment:
hi  this regex will catch false positives "fiz 233 baz 345  bar 6789" so this might not be the best one.
on the other hand the allowed NANP spec is ridiculously complex and even a subset of it spawns not very nice:
`^(\+\d{1,2}\s)?\(?\d{3}\)?[\s.-]?\d{3}[\s.-]?\d{4}$` (copied via mobile so it might vie utterly broken)

So lets meet halfway and agree that phone numbers are optional country region code + 3 groups of digits optionally separated by either single space dot or `-`. Country region code is either preceeded by `+` or `00` If its still too complex we might pair it down further as long as it will not give false positives.

Also it might be a good place to showoff named capture groups (at least for the country  code)

And we would still mention in the example that this will work only for nicely formatted numbers.